src/strategies/erc721-balance-or-delegated/README.md

```diff
@@ -0,0 +1,29 @@
+# ERC721 Balance or Delegate Strategy
+
+This strategy allows users to vote if they either hold an ERC721 NFT or have voting power delegated to them from that same NFT contract. It combines direct NFT ownership and delegated voting power to determine the final voting score for addresses, with a maximum of 1 in voting power in both cases.
```

Comment:
Hi @SheeranJL I did some review and looks like instead of a separate strategy, you could use our in built feature of voting validation https://docs.snapshot.box/user-guides/strategies/validation-strategies#voting-validation-in-space-settings 

And use `erc721`  and `contract-call` strategies inside basic validation and `ticket` strategy to give `1` voting power for eligible voters

Reply:
Thanks for that @ChaituVR i'll take a look. 

If i create a Voting Validation to achieve this, then I would no longer need to have a Strategy within the 'Strategies' section since the Voting Validation would be checking my rules - however, I don't believe you can choose NOT have at least one Strategy defined in the Strategies section. Is this correct?

Also, is it possible to test a Voting Validation via the UI before enabling it, much like you can test strategies within the 'Strategies' section?

Reply:
you need to use `basic` voting validation 
![image](https://github.com/user-attachments/assets/a18b22db-bdd2-4450-a415-386ca6f970e6)

and then with minimum score of `1`, select custom strategies 
![image](https://github.com/user-attachments/assets/765a6944-3339-4278-9d3b-99ce7752f8c6)

Reply:
Hey @ChaituVR - I played around with this Basic Vote Validation setup as per your instructions.

A summary of what I configured: 

1. Ticket strategy with 1 Vote Power 
2. Contract call strategy for **getVotes** on my NFT contract
3. ERC721 call to get holdings of NFT for a given address. 

I then also added points 2, and 3, to the main Strategies tab as their own Strategies alongside the Basic setup since I cannot remove all of the Strategies because it says you need at least one. Duplicating this seemed to have worked and assigned 1 point to all users regardless of their combination of delegation/balance most likely due to the Ticket Validation setup.

I think i'm good now unless you notice anything odd about the way I did this. Thanks so much for your help 👍

https://snapshot.org/#/hvax.eth

<img width="923" alt="Screenshot 2024-12-02 at 14 27 15" src="https://github.com/user-attachments/assets/abde4341-9cdf-4d5b-80a3-6c5ee007eea5">

Reply:
hi @SheeranJL 

- ticket strategy will give every wallet with `1` voting power, you need to remove it from the voting validation and keep the other two (unless intentional)
![image](https://github.com/user-attachments/assets/1c4be2be-abb8-4e58-a4c3-849e0ac87276)

- And you wanted to give `1` voting power to every eligible voter right? so in strategies, you should add a `ticket` strategy and remove other
![image](https://github.com/user-attachments/assets/52f3f44f-fc55-42e8-9a71-127e55052bb1)